src/lib.rs

```diff
@@ -370,12 +379,15 @@ fn write_files<'a>(files_to_write: impl Iterator<Item = (&'a Path, &'a [u8])>) {
 
 type RuleName = String;
 
-fn apply_fixes(file_contents: &mut Vec<u8>, pending_fixes: HashMap<RuleName, Vec<PendingFix>>) {
+fn apply_fixes(
+    file_contents: &mut MutRopeOrSlice,
+    pending_fixes: HashMap<RuleName, Vec<PendingFix>>,
+) {
     let non_conflicting_sorted_pending_fixes =
         get_sorted_non_conflicting_pending_fixes(pending_fixes);
     for PendingFix { range, replacement } in non_conflicting_sorted_pending_fixes.into_iter().rev()
     {
-        file_contents.splice(range, replacement.into_bytes());
+        file_contents.splice(range, &replacement);
```

Comment:
I implemented `.splice()` on `MutRopeOrSlice` which either "delegates" to this existing `Vec<u8>::splice()` or else does "splicing" on a `Rope`